core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -98,17 +98,24 @@ private[spark] class SparkSubmit extends Logging {
    * Kill an existing submission using the REST protocol. Standalone and Mesos cluster mode only.
    */
   private def kill(args: SparkSubmitArguments): Unit = {
-    new RestSubmissionClient(args.master)
-      .killSubmission(args.submissionToKill)
+    createRestSubmissionClient(args).killSubmission(args.submissionToKill)
   }
 
   /**
    * Request the status of an existing submission using the REST protocol.
    * Standalone and Mesos cluster mode only.
    */
   private def requestStatus(args: SparkSubmitArguments): Unit = {
-    new RestSubmissionClient(args.master)
-      .requestSubmissionStatus(args.submissionToRequestStatusFor)
+    createRestSubmissionClient(args).requestSubmissionStatus(args.submissionToRequestStatusFor)
+  }
+
+  /**
+   * Creates RestSubmissionClient with overridden logInfo()
+   */
+  private def createRestSubmissionClient(args: SparkSubmitArguments): RestSubmissionClient = {
+    new RestSubmissionClient(args.master) {
+      override protected def logInfo(msg: => String): Unit = printMessage(msg)
```

Comment:
doesn't this change the behavior even when the logger is initialized/working?

Reply:
When `isInterpreter = true`(repl shell) the log is initialized and working but the log level is setting to WARN due to that `RestSubmissionClient` logInfo messages are not showing which are as part of the response. This PR change effects when `isInterpreter = true` and for status/kill commands, and doesn't change the other behavior.

Reply:
this is not necessarily always the case - user can config log level easily?

Reply:
I agree, user can change this log level. If the users configure the log level as WARN or above(WARN is the default log level) then they can't see any update/status from status and kill commands. I think we cannot expect the user to configure the log level to INFO if they want to run status and kill commands with status/update. Please let me know if you have any thoughts to fix this better, I can make the changes. Thanks